src/core/public/http/fetch.test.ts

```diff
@@ -30,16 +31,90 @@ function delay<T>(duration: number) {
   return new Promise<T>(r => setTimeout(r, duration));
 }
 
+const BASE_PATH = 'http://localhost/myBase';
+
 describe('Fetch', () => {
   const fetchInstance = new Fetch({
-    basePath: new BasePath('http://localhost/myBase'),
+    basePath: new BasePath(BASE_PATH),
     kibanaVersion: 'VERSION',
   });
   afterEach(() => {
     fetchMock.restore();
     fetchInstance.removeAllInterceptors();
   });
 
+  describe('getRequestCount$', () => {
+    const getCurrentRequestCount = () =>
+      fetchInstance
+        .getRequestCount$()
+        .pipe(first())
+        .toPromise();
+
+    it('should increase and decrease when request succeeds', async () => {
+      let resolveRequest: any;
+      const response = new Promise<any>(resolve => (resolveRequest = resolve));
+      fetchMock.get('*', response);
+
+      const fetchResponse = fetchInstance.fetch('/path');
+      expect(await getCurrentRequestCount()).toEqual(1);
+
+      resolveRequest(200);
+      await fetchResponse;
+      expect(await getCurrentRequestCount()).toEqual(0);
+    });
+
+    it('should increase and decrease when request fails', async () => {
+      let resolveRequest: any;
+      const response = new Promise<any>(resolve => (resolveRequest = resolve));
+      fetchMock.get('*', response);
+
+      const fetchResponse = fetchInstance.fetch('/path');
+      expect(await getCurrentRequestCount()).toEqual(1);
+
+      resolveRequest(500);
+      await expect(fetchResponse).rejects.toThrow();
+      expect(await getCurrentRequestCount()).toEqual(0);
+    });
+
+    it('should change for multiple requests', async () => {
+      let resolveSuccessfulRequest: any;
+      const successfulResponse = new Promise<any>(resolve => (resolveSuccessfulRequest = resolve));
+      let resolveFailedRequest: any;
+      const failedResponse = new Promise<any>(resolve => (resolveFailedRequest = resolve));
```

Comment:
Shouldn't it be `const failedResponse = new Promise<any>((resolve, reject) => (resolveFailedRequest = reject));` ?

Reply:
The Fetch API doesn't throw in the 4xx or 5xx cases, only when there are network errors. Though I should add a test for that case as well 👍